### ISSUE TITLE
Applying bucket changes without terraform doesn't reflect in state after terraform refresh

### ISSUE THREAD

**Issue by srikiraju** [terraform-providers/terraform-provider-aws#2296]
### Terraform Version
Terraform v0.10.8

### Affected Resource(s)
- aws_s3_bucket

### Terraform Configuration Files
```hcl
 resource "aws_s3_bucket" "www" {
     bucket = "main_bucket"
     acl    = "private"
    replication_configuration {
       role = "${aws_iam_role.s3_replication.arn}"

       rules {
         id     = "All"
         prefix = ""
         status = "Enabled"

         destination {
           bucket        = "arn:aws:s3:::otherbucket"
           storage_class = "STANDARD"
         }
      }
    }
 }
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. Remove replication from AWS console
3. `terraform plan` - Same, plan does not detect that replication was removed
4. `terraform refresh` - Off-band changes not detected

I think a lot of the other options in s3_bucket have the same behaviour

**Comment by devonbleak** [comment]
Confirming that changing ACLs also doesn't show up - had a bucket set to public today and my terraform with `acl = "private"` showed no changes required in the plan.

**Comment by Chhed13** [comment]
I can't reproduce exactly this code, but I definitely found an issue with replication_configuration update function - it refreshes only if `replication.ReplicationConfiguration != nil` instead of set state to default value.

**Comment by Chhed13** [comment]
Added pull request for replication.
As I can see in code there is now **read** function for **ACL** implemented at all.
For **cors_rule** block - the same situation as for replication - it updates only when not nil.
And I double check **request_payer**, **versioning** and  **lifecycle_rule**. They can work not correct on remove too.

**Comment by bflad** [comment]
This has been released in [version 1.10.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1100-february-24-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by bflad** [comment]
(Since a commit already closed this ticket for replication_config, let's address each case individually)
